config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto

```diff
@@ -7,27 +7,90 @@ package org.hypertrace.config.service.v1;
+
+  // Deletes the config for the specified request
+  rpc DeleteConfig(DeleteConfigRequest) returns (DeleteConfigResponse) {}
 }
 
 message UpsertConfigRequest {
-  string resource_name = 1;  // required
-  string resource_namespace = 2;  // required - namespace with which the config resource is associated
-  google.protobuf.Value config = 3; // required - config value to be upserted
-  string context = 4; // optional - only required if config applies to a specific context
+  // required - name of the resource associated with the config
+  string resource_name = 1;
+
+  // required - namespace with which the config resource is associated
+  string resource_namespace = 2;
+
+  // required - config value to be upserted
+  google.protobuf.Value config = 3;
+
+  // optional - only required if config applies to a specific context.
+  // If empty, specified config is associated with a default context.
+  string context = 4;
 }
 
 message UpsertConfigResponse {
-  int64 config_version = 1; // the version allocated to the newly inserted configuration
+  // config value upserted into the store
+  google.protobuf.Value config = 1;
 }
 
 message GetConfigRequest {
-  string resource_name = 1;  // required
-  string resource_namespace = 2;  // required - namespace with which the config resource is associated
-  repeated string contexts = 3; // optional; order matters - specify the contexts in the order of increasing priority. Example: ["service1", "api1"]
+  // required - name of the resource associated with the config
+  string resource_name = 1;
+
+  // required - namespace with which the config resource is associated
+  string resource_namespace = 2;
+
+  // optional; order matters - specify the contexts in the order of increasing
+  // priority. Example: ["service1", "api1"].
+  // Default context is implicitly appended to the beginning of the list.
+  repeated string contexts = 3;
 }
 
 message GetConfigResponse {
+  // config value returned for the specified request
   google.protobuf.Value config = 1;
 }
+
+message GetAllConfigsRequest {
+  // required - name of the resource associated with the config
+  string resource_name = 1;
+
+  // required - namespace with which the config resource is associated
+  string resource_namespace = 2;
+}
+
+message GetAllConfigsResponse {
+  // list of config values along with the associated contexts
+  repeated ContextSpecificConfig context_specific_configs = 1;
+}
+
+message ContextSpecificConfig {
+  // context associated with the below config
+  string context = 1;
+
+  // config value which applies to the above context
+  google.protobuf.Value config = 2;
+}
+
+message DeleteConfigRequest {
+  // required - name of the resource associated with the config
+  string resource_name = 1;
+
+  // required - namespace with which the config resource is associated
+  string resource_namespace = 2;
+
+  // required - for default context, use "DEFAULT-CONTEXT"
```

Comment:
Could you remove the default context comment - that's an implementation detail - we explicitly decided not to support deleting this.

Reply:
Removed it from the comments. But should we explicitly check if the context is set to `DEFAULT-CONTEXT`, then deny that request?

Reply:
That's not a bad idea - up to you